canon/src/main/scala/cats/canon/ApplicativeFromMonad.scala

```diff
@@ -0,0 +1,7 @@
+package cats
+package canon
+
+trait ApplicativeFromMonad[F[_]] extends Applicative[F] with FunctorFromApplicative[F] { self: Monad[F] =>
+  def ap[A, B](fa: F[A])(ff: F[A => B]): F[B] =
+    flatMap(ff)(f => flatMap(fa)(a => pure(f(a))))
+}
```

Comment:
This works. If the purpose is not apparent I will go with your approach, the only reason to go with self-types here was because I thought they expressed the intent better.

I read the signature 
```scala
trait FunctorFromApplicative[F[_]] extends Functor[F] { self: Applicative[F] =>
```

as "This trait provides a Functor[F] when mixed into an Applicative[F]".

Conversely, I don't believe the signature
```scala
trait CanonicalApplicative[F[_]] extends Applicative[F] {
```
provides as much information about how it is intended to be used (as a mix-in).

I expected the mix-in approach to scale better when you want to pick-and-choose which defaults you want -- for example

```scala
implicit val optionInstance =
  new Monad[Option] with Traverse[Option] with FunctorFromTraverse[Option] {
    ...
  }
```
and
```scala
implicit val optionInstance =
  new Monad[Option] with Traverse[Option] with FunctorFromApplicative[Option] {
    ...
  }
```
should have different implementations of map.

Whether this is desirable is unclear.